students/Dkuchan/ClassProject/PythonClassProject.py

```diff
@@ -0,0 +1,625 @@
+                    matchlist[i].penalties = line[11]
+
+    print()
+    print()
+    print("File has been read.")
+    print()
+    print()
+    createteamlist()
+    primaryUI()
+
+
+def outputmatchdata(matchnumber):
+    for i in range(0, len(matchlist)):
+        if matchlist[i].matchnumber == matchnumber:
+            print("")
+
+
+def produceinsights(teamnum):
+    # This function calls multiple functions to provide a report
+    # containing statistical outcomes that are applicable
+    # to the specified team.
+    pass
+
+
+def doesteamexist(team):
+    # looks through a list of teams and checks whether the team exists.
+    existance = False
+    for i in range(0, len(teamlist)):
+        if team == str(teamlist[i]):
+            existance = True
+    return existance
+
+
+def duplicatecheck(team, array):
+    # takes a team and an array, checks if it occurs more than once
+    duplicatestat = False
+    if array.count(team) >= 1:
+        duplicatestat = True
+    return duplicatestat
+
+def executemonte(matchsetuparray, runs=100000, factor = .2, rangemin = -1, rangemax = 1):
+    # this functions executes a monte carlo run.
+    # it is nominally called from the setupmonteUI fnc.
+    # currently this monte carlo ONLY works on gear scores
+    grossresults = []   #this will be a memory hog try to think of how to use a generator
+    #create array with team's average score
+    bluewin = 0
+    tie = 0
+    percentconversion = runs / 100
+    for i in range(0, runs):
+        scores = [None, None, None, None, None, None]
+        for j in range(0, len(scores)):
+            roll = random.randint(rangemin, rangemax)  # creates a random number between the range bounds
+            scores[j] = roll * teamdict[matchsetuparray[j]].averagegears  # adds roll value to average 
+        for k in range(0, len(scores)):  # prevents a negative score
+            if scores[k] <= 0:
+                scores[k] = 0
+        bluescore = scores[0] + scores[1] + scores[2]   #needed to sum half the array each
+        redscore = scores[3] + scores[4] + scores[5]
+        if bluescore > redscore:    # who won?
+            bluewin += 1
+        elif bluescore == redscore:  # if nobody won
+            tie += 1
+        print("Red: {}, Blue: {}".format(redscore, bluescore))
+    blueresults = bluewin / percentconversion
+    tieresults = tie / percentconversion
+    print("Blue Wins {} percent of the time.".format(blueresults))
+    print("Red Wins {} percent of the time.".format((runs - bluewin - tie) / percentconversion))
+    print("Game ties {} percent of the time.".format(tieresults))
+    primaryUI()
+
+        #montescores.append(MonteResult(bluescore, redscore, i)) #creates an instance of monte match.monte
+        #dump detailed result file.
+
+
+def showteamdata(teamnumber):
+    print()
+    print()
+    print(str(teamdict[teamnumber]))
+    print("Matches Played: {}".format(teamdict[teamnumber].nummatches))
```

Comment:
probably a goo didea to create a variable:

`team = teamdict[teamnumber]`

then use `team` everywhere else...